### PR TITLE
Polynomial serialization, part 1: writing

### PR DESCRIPTION
**Pull request by pleroy**
This requires a change in the way that quantities exponents are serialized, because we need to support large negative exponent.  Hopefully this won't be save-incompatible as we probably never used large positive exponents.

### CODE REVIEW

File: quantities/quantities_body.hpp [hunk 1]
```diff
@@ -202,6 +201,22 @@ struct NthRootGenerator<
...
+}
+
+constexpr std::int64_t ExponentSerializer::Representation(
+    std::int64_t const exponent,
+    std::int64_t const position) {
+  // For exponents in [-16, 7] this returns the representations
+  // 0x10, 0x11, ... 0x00, ... 0x07.  For exponents in [-24, -17] this returns
+  // the representations 0x08, 0x09, ... 0x0F.  The latter used to be reserved
+  // for exponents in the range [8, 15] but we believe that we never used them,
+  // and with polynomials in the monomial base we need large negative exponents.
```
Comment (eggrobin) [inline quantities/quantities_body.hpp]: basis

File: quantities/quantities_body.hpp [hunk 2]
```diff
@@ -9,11 +9,34 @@
...
+  // 0-based position of the dimension in the representation.
+  static constexpr std::int64_t Representation(
+      std::int64_t exponent,
+      std::int64_t position);
+
+ private:
+  static constexpr std::int64_t min_exponent = -24;
+  static constexpr std::int64_t max_exponent = 7;
+  static constexpr std::int64_t exponent_mask = 0x1F;
+  static constexpr std::int64_t exponent_bits = 5;
```
Comment (eggrobin) [inline quantities/quantities_body.hpp]: Do we know why it is 5 and not 6? if so, document the historical reasons; perhaps we should migrate to 8 bits in the future.

File: serialization/numerics.proto [hunk 3]
```diff
@@ -39,3 +39,25 @@ message DoublePrecision {
   required Value value = 1;
   required Error error = 2;
 }
+
+// Polynomials follow.
```
Comment (eggrobin) [inline serialization/numerics.proto]: Why this comment?

File: quantities/quantities_body.hpp (hunk 2 above)
Comment (pleroy) [reply quantities/quantities_body.hpp]: I have no idea.

File: quantities/quantities_body.hpp (hunk 1 above)
Comment (pleroy) [reply quantities/quantities_body.hpp]: Done.

File: serialization/numerics.proto (hunk 3 above)
Comment (pleroy) [reply serialization/numerics.proto]: We do that in the other protos to indicate that all the subclasses/extensions are grouped together.

**Pull request merged**